### DISCUSSION TITLE
Using typescript files for translations

### DISCUSSION THREAD

**Discussion by Daidalos117** [i18next/next-i18next#2010] [Q&A]
Hi!

I want to use typescript files for as translation source. Mainly because I want to leverage enums which are already in place, like so:
```
const translations = {
    [UserTypeEnum.USER]: 'User'
}
```

I thought about setting it  as the resource in config file, but that doesn't work mainly because next config files doesn't support typescript. I am surprised I didn't find any mentions about this. 
What would you recommend?

**Comment by adrai** [comment]
Keep using the i18next standard... JSON files

**Reply by Daidalos117** [reply]
Well i18next supports passing objects in init function. Problem here is that we need to pass it in the config file. JSON is limited in numerous ways. Even tho I can clearly see some advantages for some projects, in my use case it's just not the best way. 
How would you use enum in the JSON file?
